### DISCUSSION TITLE
how to use "pulseIn" function for Raspberry pi

### DISCUSSION THREAD

**Discussion by grekichi** [WiringPi/WiringPi#297] [General]
Hi there !
I want to reuse Arduino program by raspberry pi.
But "WiringPi" don't prepare "pulseIn" function, right ?
So, how should I do to use same function for my project ?
If you have any advice, please tell me.

Thanks in advance.

**Comment by nohren** [comment]
@mwallner @mstroh76 @grekichi I just made a commit off of develop to add this function. I saw its use in the HC-SR04 ultrasonic sensor. [link](https://howtomechatronics.com/tutorials/arduino/ultrasonic-sensor-hc-sr04/).  How do I get access to push a feature branch?

**Reply by Andy2No** [reply]
@nohren,

I think you just have to go to the Pull requests page and click "New Pull Request":
https://github.com/WiringPi/WiringPi/pulls
